### PR TITLE
Remove javaCrash failure condition and detect daemon expiring message

### PR DESCRIPTION
**Pull request by blindpirate**
Sometimes there are JVM crashing and OOM messages in the test log,
which fail the TeamCity build. In these cases, we don't want the build
to fail because they will be handled by test-retry.

However, we do want to capture the OOM in build daemon, let's try
if we can monitor "Expiring Daemon because JVM heap space is exhausted"
message in build log.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle cancel

**Comment by blindpirate** [comment]
@bot-gradle test this

File: .teamcity/src/main/kotlin/common/extensions.kt [hunk 1]
```diff
@@ -128,6 +128,14 @@ fun BuildType.applyDefaultSettings(os: Os = Os.LINUX, arch: Arch = Arch.AMD64, b
                 stopBuildOnFailure = true
             }
         }
+        add {
+            failOnText {
+                conditionType = BuildFailureOnText.ConditionType.CONTAINS
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/common/extensions.kt:133]: Does this include test output?

File: .teamcity/src/main/kotlin/configurations/FunctionalTest.kt [hunk 2]
```diff
@@ -52,14 +52,6 @@ class FunctionalTest(
         extraSteps = extraBuildSteps,
         preSteps = preBuildSteps
     )
-
-    if (testCoverage.testType == TestType.soak || testTasks.contains("plugins:")) {
-        failureConditions {
-            // JavaExecDebugIntegrationTest.debug session fails without debugger might cause JVM crash
-            // Some soak tests produce OOM exceptions
-            javaCrash = false
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/configurations/FunctionalTest.kt:60]: I think `javaCrash` = false needs to stay here to disable crash detection for all functional tests, right?

File: subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/server/health/LowHeapSpaceDaemonExpirationStrategyTest.groovy [hunk 3]
```diff
@@ -17,10 +17,12 @@
 package org.gradle.launcher.daemon.server.health
 
 import org.gradle.launcher.daemon.server.expiry.DaemonExpirationResult
+import spock.lang.Ignore
 import spock.lang.Specification
 
 import static org.gradle.launcher.daemon.server.expiry.DaemonExpirationStatus.GRACEFUL_EXPIRE
 
+@Ignore
```
Comment (wolfs) [inline subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/server/health/LowHeapSpaceDaemonExpirationStrategyTest.groovy:25]: Why ignore this test?

**Comment by wolfs** [comment]
The problem here is not that we do fail the build when we detect the OOM message. The problem is that now that we replay test results, even though the next build doesn't fail TC still detects the OOM message and then fails the build. I am not convinced we want to detect the expiring daemon message, since we also have our own Grafana dashboard for healthy/unhealthy daemons which should detect this as well, right?

**Review by wolfs — CHANGES_REQUESTED** [review]

**Pull request closed without merging**